polygeist/tools/cgeist/driver.cc

```diff
@@ -369,8 +369,13 @@ static int optimize(mlir::MLIRContext &Ctx,
     // operations to be inlined.
     if (RaiseToAffine)
       OptPM.addPass(mlir::createLowerAffinePass());
-    PM.addPass(sycl::createInlinePass(sycl::InlineMode::Simple,
-                                      /* RemoveDeadCallees */ true));
+    if (OmitOptionalMangledFunctionName) {
```

Comment:
Question: if `OmitOptionalMangledFunctionName` is false and `InlineSYCLMethodOps` is true can the inliner correctly inline SYCLMethod operations ? From the comment it would seem it cannot because the required `MangledFunctionName` attribute would ber missing.

Reply:
The message was misleading. The pass we're running here converts `SYCLMethodOps` to `sycl.call`, so the inliner will have all of the info available to build the CFG.

Reply:
ok good